diff --git a/clang/lib/Driver/ToolChains/CommonArgs.cpp b/clang/lib/Driver/ToolChains/CommonArgs.cpp
index 37ec7346857..d592609b350 100644
--- a/clang/lib/Driver/ToolChains/CommonArgs.cpp
+++ b/clang/lib/Driver/ToolChains/CommonArgs.cpp
@@ -150,6 +150,13 @@ void tools::AddLinkerInputs(const ToolChain &TC, const InputInfoList &Inputs,
   if (!TC.isCrossCompiling())
     addDirectoryList(Args, CmdArgs, "-L", "LIBRARY_PATH");
 
+  if (!TC.isCrossCompiling()) {
+    std::string lib_dir = llvm::sys::path::parent_path(TC.getDriver().Dir).str() + "/lib" + CLANG_LIBDIR_SUFFIX;
+    CmdArgs.push_back(Args.MakeArgString("-L" + lib_dir));
+    CmdArgs.push_back("-rpath");
+    CmdArgs.push_back(Args.MakeArgString(lib_dir));
+  }
+
   for (const auto &II : Inputs) {
     // If the current tool chain refers to an OpenMP or HIP offloading host, we
     // should ignore inputs that refer to OpenMP or HIP offloading devices -
